utils.py

```diff
@@ -185,10 +200,38 @@ def generate_sized_crashes(size):
     payload, headers = multipart_encode(crash_payload)
     base_size = len(payload)
 
-    # Create a "dump file" which is really just a bunch of 'a' such that
-    # the entire payload is equal to size
+    # Create a "dump file" which is really just a bunch of random
+    # characters such that the entire payload is equal to size
     dumps['upload_file_minidump'] = 'a' * (size - base_size)
 
+    if not compressed:
+        # This is the easy case, so we just do it and move on
+        return raw_crash, dumps
+
+    # This is the hard case. We need to generate a payload that when compressed
+    # is of the right size. So we loop.
+    def get_char():
+        return random.choice(string.ascii_letters)
+
+    compressed_size = len(compress(payload))
+    while compressed_size != size:
+        if compressed_size > size:
+            # If we're over, then drop the first 10 characters
+            dumps['upload_file_minidump'] = dumps['upload_file_minidump'][10:]
+
+        else:
+            # If we're under, then add a bunch of characters in a way that
+            # tries to "bisect" the distance we still need to cover because we
+            # want to get on with living our lives
+            num = int((size - compressed_size) / 2) + 1
+            dumps['upload_file_minidump'] += ''.join(
+                [get_char() for i in range(num)]
+            )
+
+        crash_payload = assemble_crash_payload(raw_crash, dumps)
+        payload, headers = multipart_encode(crash_payload)
+        compressed_size = len(compress(payload))
```

Comment:
This while loop is goofy, but I couldn't think of a better way. It's tricky because compression operates on sequences, not single bytes, so it's hard to predict how changes to `upload_file_minidump` affect the resulting compressed size. This does a "bisect" kind of approach where it adds "half" the size of random characters each iteration. That makes it much faster than doing them one at a time.

There are probably better ways to do it, but this works and with the memoization it only happens once per test and with the size check I added in `post_crash`, the payload size is correct or it fails.